core/src/test/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1238,7 +1238,7 @@ public void restartRandomDataNode(RestartCallback callback) throws Exception {
     /**
      * Restarts a random node in the cluster and calls the callback during restart.
      */
-    private void restartRandomNode(Predicate<NodeAndClient> predicate, RestartCallback callback) throws Exception {
+    public void restartRandomNode(Predicate<NodeAndClient> predicate, RestartCallback callback) throws Exception {
```

Comment:
You can use the RestartCallback clear data method like here: https://github.com/elastic/elasticsearch/blob/master/core/src/test/java/org/elasticsearch/gateway/RecoveryFromGatewayTests.java#L253